### DISCUSSION TITLE
serialEvent not handled

### DISCUSSION THREAD

**Discussion by asharect** [feilipu/Arduino_FreeRTOS_Library#7] [Q&A]
I used arduino example "SerialEvent", it was fine. But when I include Arduino_FreeRTOS, the serialEvent didn't work any more,  how to fix it? rewrite avr ISR? but I noticed that there's no configMAX_SYSCALL_INTERRUPT_PRIORITY option in freertosconfig.h, it seems that the uart interupt always is blocked by task, even idle task.

**Comment by feilipu** [comment]
I have tried to replicate the issue you describe, but I can't.

See below screenshot.
![screenshot from 2016-08-30 19-55-07](https://cloud.githubusercontent.com/assets/3955592/18084908/47c3c1ae-6eec-11e6-8b19-030c4ecf8b69.png)

The AVR ATmega is a very simple MCU. There is no interrupt nesting system (except intrinsic ordering) available. The [configMAX_SYSCALL_INTERRUPT_PRIORITY](http://www.freertos.org/a00110.html) option is not relevant to an ATmega port of FreeRTOS.

Please, check that you have the latest 8.2.3-16 Release of the Arduino_FreeRTOS library installed.

**Comment by joe31093** [comment]
I am not sure if this helps.  I recently installed 8.2.3 (greiman) on my Teensy 3.6.  The serial event is not being call.  I have two tasks running.  A heartbeat task which turns on off the LEDs.  Another task waiting on a sem give in the serial event function so in can process the text.  I have a print statement in the serial event and it is not being printed.  I ran a similar example of the code you have above and the serial event worked fine.  I am guessing running the scheduler affected the system interrupts.
